share/completions/git.fish

```diff
@@ -67,16 +67,24 @@ function __fish_git_remotes
 end
 
 function __fish_git_modified_files
+    set -l matching_files *(commandline -ct)*
+    if not set -q matching_files[1]
+        return
+    end
     # git diff --name-only hands us filenames relative to the git toplevel
     set -l root (command git rev-parse --show-toplevel ^/dev/null)
     # Print files from the current $PWD as-is, prepend all others with ":/" (relative to toplevel in git-speak)
     # This is a bit simplistic but finding the lowest common directory and then replacing everything else in $PWD with ".." is a bit annoying
-    string replace -- "$PWD/" "" "$root/"(command git diff --name-only $argv ^/dev/null) | string replace "$root/" ":/"
+    string replace -- "$PWD/" "" "$root/"(command git -C $root diff --name-only $argv -- $matching_files ^/dev/null) | string replace "$root/" ":/"
```

Comment:
Unfortunately, it seems like git uses the `-C` path as the working directory completely, so this needs to be `$PWD/$matching_files`.

Reply:
E.g. go to $fish_shell_repo/share/completions, modify git.fish, and see the following:

```fish
git -C $fish_shell_repo diff -- git.fish
``` 

It'll print nothing.